### ISSUE TITLE
Router Base Attribute does break hash links

### ISSUE THREAD

**Issue by HannesOberreiter** [nuxt/content#376]
### Version
@nuxt/content: 1.5.3
nuxt: 2.14.0

### Reproduction Link
https://codesandbox.io/s/autumn-https-x3oxj?file=/nuxt.config.js

You need do change the demo browser window to the subfolder `https://x3oxj.sse.codesandbox.io/test/`

### Steps to reproduce

Set base attribute inside `nuxt.config.js`

```
  router: {
    base: "/test"
  },
```

Go to the `about` page and click on the footnote. You will the URL changing to the hash but removing the current page `about`. This does also happening with the other types of hash routes on the same page, eg. headings or content links.

### What is Expected?

Hash should be added to current page url.

### What is actually happening?

Hash will remove the current Url and will be added to the base.

PS: On discord I made a video and gif of the problem, because I'm trying to deploy my nuxt-content page inside a subfolder. 

Cheers
Hannes

**Comment by b1rdhous3** [comment]
Any Updates on this Topic? I have the same issue.

**Comment by Atinux** [comment]
Hi there,

When using `router.base`, Nuxt adds a `<base href="/test/">`.

Sadly it looks like a known issue from the browser: https://rogerkeays.com/blog/using-base-href-with-anchors

You can use a Nuxt hook to remove the `<base>` tag, actually it fixes your issue but may introduces somes
 issues (you will have to check to make sure your websites works perfectly without it):

```js
// nuxt.config.js
export default {
hooks: {
    "vue-renderer:ssr:templateParams": function (params) {
      params.HEAD = params.HEAD.replace('<base href="/test/">', "");
    }
  }
}
```

Working sandbox: https://codesandbox.io/s/green-meadow-tquyq?file=/nuxt.config.js:838-1027

**Comment by HannesOberreiter** [comment]
@Atinux thanks that you looked into the issue and the workaround. It seems to work for me. I'm closing this issue as the problem is browser specific.

Don't know if it is a good idea to write a remark about this in the docs? [en/guides/configuration-glossary/configuration-router.md](https://github.com/nuxt/nuxtjs.org/blob/master/content/en/guides/configuration-glossary/configuration-router.md)

Cheers
Hannes

**Comment by b1rdhous3** [comment]
Nice, thank you! @Atinux 

For information: I noticed that this problem also affects the "on this site" anchor links on the right of a page. For e.g. in the docs Theme. 
So maybe something some important than footnotes. 
Nevertheless my browser could jump to the "broken" anchors at the "on this site" links, but not at footnotes. Links looks the same.

**Comment by HannesOberreiter** [comment]
@b1rdhous3 yes forgot to mention this, but it is happening for all hash links. eg. in docs theme if you click the headers and the links in the ToC.

**Comment by b1rdhous3** [comment]
So I found one Side Effect to @Atinux suggestion: 
If I try to link an image in a .md file within a subdirectory the image can not be found. It was expected at following location: 
http://localhost:3000/basedir/subdir/OCR.png

To show the image I have to link it this way: 
<img src="../image.png" />

If the .md is in no subdirectory everything is working fine.

**Comment by dumptyd** [comment]
This https://github.com/nuxt/content/issues/376#issuecomment-702193217 almost worked for me, except for the images as @b1rdhous3 pointed out.

This is how I solved it:

```bash
yarn add rehype-urls
```
 In `nuxt.config.js`

```js
const path = require('path');
const baseURL = '/blog/';

export default {
  content: {
    markdown: {
      rehypePlugins: [
        ['rehype-urls', function addBaseToImages(url) {
          if (!baseURL) return;
          // replace '/images/' with whatever directory you're keeping them in
          if (url.href && url.href.startsWith('/images/')) return path.join(baseURL, url.href);
        }]
      ]
    }
  }
};
```

Nuxt masks this issue by adding a redirect rule from / from base, so unless you build and test the generated bundle you won't find this issue until it's deployed to prod. Might be a good idea to not add those redirects, but that's for upstream.
